### ISSUE TITLE
Provide better default scheduling behaviour for Google Cloud preemptible instances in Instance Templates

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-google#107]
_This issue was originally opened by @danielcompton as hashicorp/terraform#15055. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
Terraform v0.9.6

### Affected Resource(s)
Please list the resources as a list, for example:
- google_compute_instance_template

### Terraform Configuration Files
```hcl
resource "google_compute_instance_template" "instance_template" {
  name_prefix = "my-dev-template-"
  description = "template description"

  tags = ["dev"]

  instance_description = "desc "
  machine_type         = "n1-standard-1"
  can_ip_forward       = false

  scheduling {
    preemptible       = true
  }

  // Create a new boot disk from an image
  disk {
    source_image = "debian-cloud/debian-8"
    auto_delete  = true
    boot         = true
  }

  network_interface {
    network = "default"
  }

  service_account {
    scopes = ["sql-admin", "servicecontrol", "service-management", "logging-write", "monitoring-write", "trace-append", "devstorage-rw"]
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

### Log output
https://gist.github.com/danielcompton/6f29b9ae9e0507be2a1c81849a234591

### Expected Behavior
The instance template using a preemptible VM should have been created.

### Actual Behavior
I got an error 

```
Error 400: Invalid value for field 'resource.properties.scheduling.preemptible': 'true'. Scheduling must have preemptible be false when AutomaticRestart is true., invalid
```

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
This is happening because `scheduling.automatic_restart` defaults to true, which is useful and matches the behaviour for standard instances, but isn't a valid option for preemptible instances. This may also apply to standard compute instances, but I haven't checked.

A few possible options I can see would be to remove the automatic_restart default entirely and let it be computed, use DefaultFunc, or maybe something else.

I also noticed that the base `automatic_restart` value causes resources to be deleted and then recreated if you have specified `scheduling.automatic_restart`, as it defaults to true, which is then treated as a change to the resource. Providing an explicit false for both of them works, but then you get warnings about using a deprecated attribute.

**Comment by selmanj** [comment]
Possibly relates to #142. After #142 is merged in, we will support being able to rely on a server-specified default. I believe (but haven't yet verified) that when AutomaticRestart is not explicitly specified but Preemptible is, AutomaticRestart will then be set to false which will make this slightly less painful.

**Comment by selmanj** [comment]
Actually looking at the original issue, setting the field to computed seems like a much better solution than trying to guess it ourselves.

**Comment by selmanj** [comment]
So, unfortunately I wasn't able to use a `DefaultFunc` currently as it's not possible to examine other values. However I did fix the issue where `automatic_restart` and `scheduling.automatic_restart` would both need to be specified at the same time (mostly by removing `automatic_restart`). You'll still have to explicitly specify `scheduling.automatic_restart` to `false` for preemptible instances.

Hopefully this fixes things for you; please reopen/file a new issue if you find any additional issues.

**Comment by danielcompton** [comment]
Thanks a lot, I'll take a look when I upgrade to the new provider version. Thanks!

**Comment by chrissound** [comment]
I also seem to run into this same issue with the following:

```
resource "google_compute_instance" "default" {
  name         = "gitlab"
  machine_type = "n1-highmem-2"
  zone         = "us-central1-a"
  project      = "${var.gcp_project}"
  allow_stopping_for_update = true
  scheduling {
    preemptible = true
  }

  tags = ["gitlab"]

  boot_disk {
    initialize_params {
      image = "debian-cloud/debian-9"
    }
  }

  metadata = {
    sshKeys = "${var.gce_ssh_user}:${file(var.gce_ssh_pub_key_file)}"
  }

  network_interface {
    network = "default"

    access_config {
      // Ephemeral IP
    }
  }
  # metadata_startup_script = "echo hi > /test.txt"

  service_account {
    scopes = ["userinfo-email", "compute-ro", "storage-ro"]
  }
}
```

terraform version
Terraform v0.12.9
